activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -63,8 +63,8 @@ def locking_enabled? #:nodoc:
 
         def increment_lock
           lock_col = self.class.locking_column
-          previous_lock_value = send(lock_col).to_i
-          send(lock_col + "=", previous_lock_value + 1)
+          previous_lock_value = _read_attribute(lock_col)
```

Comment:
We can remove `to_i` because `send(lock_col)`(reader) returns "typecast_value". Also, we have test cases with `nil`. When i was investigating i found new bug and opened new PR to fix that, see #28914